uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -735,21 +748,35 @@ define([
 
       /**
        * Initialize from and to dates, day types and contact
+       *
+       * @return {Promise}
        */
       function initDates() {
+        var deferred = $q.defer();
+
         if (canEdit()) {
           var attributes = _.cloneDeep(vm.leaveRequest.attributes());
 
           vm.uiOptions.fromDate = convertDateFormatFromServer(vm.leaveRequest.from_date);
+
           vm.onDateChange(vm.uiOptions.fromDate, 'from')
             .then(function () {
               //to_date and type has been reset in above call so reinitialize from clone
               vm.leaveRequest.to_date = attributes.to_date;
               vm.leaveRequest.to_date_type = attributes.to_date_type;
               vm.uiOptions.toDate = convertDateFormatFromServer(vm.leaveRequest.to_date);
-              vm.onDateChange(vm.uiOptions.toDate, 'to');
+              vm.onDateChange(vm.uiOptions.toDate, 'to')
+                .then( function () {
+                  //reolve only after both from and to day types are also set
+                  deferred.resolve({});
```

Comment:
Do you need to pass any parameter at all if you really just want to `resolve` the promise?